src/mds/MDCache.cc

```diff
@@ -8471,7 +8485,7 @@ int MDCache::path_traverse(MDRequestRef& mdr, MDSContextFactory& cf,
 	// directory isn't complete; reload
         dout(7) << "traverse: incomplete dir contents for " << *cur << ", fetching" << dendl;
         touch_inode(cur);
-        curdir->fetch(cf.build(), path[depth]);
+        curdir->fetch(path[depth], snapid, cf.build());
```

Comment:
When the memory is sufficient, the fetch key is required every time a file is created and must be retried, because the judgment condition of the incomplete directory and the Bloom filter will be invalid. I think it is better to set a judgment condition when the memory is enough to fetch the entire directory.

Reply:
Fetch single key works for a long time in our production environment and has good Performance.